src/main/java/org/apache/commons/imaging/formats/tiff/TiffImagingParameters.java

```diff
@@ -119,38 +126,64 @@ public void setExif(TiffOutputSet exif) {
         this.exif = exif;
     }
 
-    public Integer getSubImageX() {
-        return subImageX;
+    /**
+     * Sets parameters for performing a partial read operation on an image. This
+     * method is useful for reducing memory and run-time overhead when accessing
+     * large source images.
+     *
+     * @param x pixel coordinate of the upper-left corner of the source image
+     * @param y pixel coordinate of the upper-left corner of the source image
+     * @param width width of the image subset to be read
+     * @param height height of the image subset to be read
+     */
+    public void setSubImage(int x, int y, int width, int height) {
+        if(x<0 || y<0){
+            throw new IllegalArgumentException(
+                    "Invalid sub-image specification: negative x and y values not allowed");
+        }
+        if (width <= 0 || height <= 0) {
+            throw new IllegalArgumentException(
+                    "Invalid sub-image specification width and height must be greater than zero");
```

Comment:
I think similar checks in the code are throwing `ImageReadException` when validating input parameters (either user-provided, or from image metadata). Maybe we should use that here too?

Reply:
Thanks.   I spent a lot of time thinking about whether Imaging should use the IllegalArgumentException or ImageReadException in this case and couldn't really decide. So I have no problem changing it to ImageReadException.  It used to be that the parameters were evaluated when the image was read, so ImageReadException was unambiguously the right choice. But now they are evaluated earlier, when the imaging parameters are specified. So IllegalArgumentException might be more logical.  But, as you said, it requires logic to catch two different kinds of exceptions.  So it would be less convenient for the application developer.

I am planning on taking your suggestion and changing it to ImageReadException (or, ImageWriteException, where appropriate).

Gary